lib/cc/analyzer/container.rb

```diff
@@ -94,6 +96,7 @@ def docker_run_command(options)
       def read_stdout(out)
         Thread.new do
           out.each_line(@output_delimeter) do |chunk|
+            @stdout_io.write(chunk)
```

Comment:
I think we need to actually avoid always storing stdout for all containers. The Engines' stdout is completely useless and can be _crazy_ big, already causing memory issues across our infrastructure.

My original thought is that we could default `@on_output` to write to a `StringIO`, like you have here (instead of the current just-ignore-it lamba).

Then engines would continue to override it to read their results (and not bloat up a big `@stdout_io` object) but non-engines would have access to that output in the container data passed to the lifecycle methods.